@xen-orchestra/web/src/components/DonutWithLegends.vue

```diff
@@ -0,0 +1,86 @@
+<template>
+  <div class="donut-with-legends">
+    <DonutChart :segments="donutSegments" :icon :max-value="maxValue" />
+    <div class="legends-and-title">
+      <slot />
+      <ul class="legends">
+        <UiLegend
+          v-for="(segment, index) in segments"
+          :key="index"
+          :color="segment.color"
+          :value="segment.value"
+          :unit="segment.unit"
+          :tooltip="segment.tooltip"
+        >
+          {{ segment.label }}
+        </UiLegend>
+      </ul>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import type { DonutSegment } from '@core/types/donut-chart.type'
+import type { LegendColor } from '@core/types/ui-legend.type'
+import DonutChart from '@core/components/DonutChart.vue'
+import UiLegend from '@core/components/UiLegend.vue'
+import type { IconDefinition } from '@fortawesome/fontawesome-common-types'
+import { computed } from 'vue'
+
+const props = defineProps<{
+  segments: {
+    label: string
+    value: number
+    unit?: string
+    color: LegendColor
+    tooltip?: string
+  }[]
+  icon?: IconDefinition
+  maxValue?: number
+}>()
+
+defineSlots<{
+  default: () => void
+}>()
+
+const donutSegments = computed(() => {
+  if (props?.segments.length === 0) {
+    return []
+  }
+  return props.segments
+    .filter(segment => segment.color !== 'dark-blue')
+    .map(segment => {
+      let color
+      if (segment.color === 'disabled' || segment.color === 'default') {
+        color = 'unknown'
+      } else {
+        color = segment.color
+      }
+
+      return {
+        value: segment.value,
+        color,
+      }
+    }) as DonutSegment[]
+})
```

Comment:
1. `props` always exists, the `?` is useless.
2. No need to check for `segments.length`. If `props.segments` is empty, then `.filter.map` will return an empty array anyway.
3. It makes no sense to accept `dark-blue` as segment color if these segments are then completely excluded from the result.

```suggestion
const donutSegments = computed(() =>
  props.segments.map(segment => ({
    value: segment.value,
    color: segment.color === 'unknown' ? 'disabled' : segment.color,
  } as DonutSegment))
)
```